pkg/volume/testing/testing.go

```diff
@@ -1530,3 +1533,132 @@ func (f *fakeVolumeHost) SetKubeletError(err error) {
 func (f *fakeVolumeHost) WaitForCacheSync() error {
 	return nil
 }
+
+type FakeOperationStartTimeCache struct {
+	cache   sync.Map
+	deleted sync.Map
+}
+
+type fakeOperationTimestamp struct {
+	plugin             string
+	operation          string
+	countAddIfNotExist int
+	countDelete        int
+	countUpdatePlugin  int
+}
+
+func newFackOperationTimestamp(plugin, op string) *fakeOperationTimestamp {
+	return &fakeOperationTimestamp{
+		plugin:    plugin,
+		operation: op,
+	}
+}
+
+func NewFakeOperationStartTimeCache() util.OperationStartTimeCache {
+	return &FakeOperationStartTimeCache{
+		cache:   sync.Map{},
+		deleted: sync.Map{},
+	}
+}
+
+func (c *FakeOperationStartTimeCache) AddIfNotExist(key, pluginName, operationName string) {
+	ts := newFackOperationTimestamp(pluginName, operationName)
+	ts.countAddIfNotExist = 1
+	obj, loaded := c.cache.LoadOrStore(key, ts)
+	if loaded {
+		oldTs, ok := obj.(fakeOperationTimestamp)
+		if ok {
+			oldTs.countAddIfNotExist += 1
```

Comment:
There's a race here. It might be easier to just use a regular map with a mutex, it's easier to reason about and we don't need throughput optimizations for testing.

Reply:
used a regular map + rwlock as suggested. thnx